packages/core/database/lib/index.js

```diff
@@ -49,6 +50,43 @@ class Database {
     return this.entityManager.getRepository(uid);
   }
 
+  async transaction(cb) {
+    const notNestedTransaction = !transactionCtx.get();
+    const trx = notNestedTransaction ? await this.connection.transaction() : transactionCtx.get();
+
+    async function commit() {
+      if (notNestedTransaction) {
+        await trx.commit();
+      }
+    }
+
+    async function rollback() {
+      if (notNestedTransaction) {
+        await trx.rollback();
+      }
+    }
+    if (!cb) {
+      return {
+        commit,
+        rollback,
+        get() {
+          return trx;
+        },
+      };
+    }
+
+    return transactionCtx.run({ trx, commit, rollback }, async () => {
```

Comment:
I must admit I'm a bit confused by the fact you're creating a `trx` variable, then a `commit` & `rollback` function, and then get three variables with the same name from the `transactionCtx.run` callback. Could you explain the process, please?

Reply:
oh no that's just wrong it should be in cb({ trx, commit, rollback }), will fix it